### PR TITLE
Rename the ErrorHandler component to ErrorCatcher

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Review by lyrixx — DISMISSED** [review]
You forgot to update composer.json for other components: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/HttpKernel/composer.json#L20

**Comment by fabpot** [comment]
@lyrixx fixed, thank you

File: src/Symfony/Component/ErrorCatcher/composer.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-    "name": "symfony/error-handler",
+    "name": "symfony/error-catche",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/ErrorCatcher/composer.json]: catcher

File: src/Symfony/Component/HttpKernel/composer.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{
+f{
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/composer.json]: typo

File: src/Symfony/Component/ErrorCatcher/composer.json (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/ErrorCatcher/composer.json]: fixed

File: src/Symfony/Component/HttpKernel/composer.json (hunk 2 above)
Comment (fabpot) [reply src/Symfony/Component/HttpKernel/composer.json]: I HATE my Mac. The keyboard is BROKEN, like many other fellows.

**Review by Simperfit — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/error_catcher.xml [hunk 3]
```diff
@@ -7,31 +7,31 @@
     <services>
         <defaults public="false" />
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/error_catcher.xml]: to be removed (spotted by @Tobion )

Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/error_catcher.xml]: done

File: src/Symfony/Component/Messenger/composer.json [hunk 4]
```diff
@@ -38,8 +38,7 @@
     },
     "conflict": {
         "symfony/event-dispatcher": "<4.3",
-        "symfony/http-kernel": "<4.4",
-        "symfony/debug": "<4.4"
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/composer.json]: we should keep this line

File: src/Symfony/Bundle/TwigBundle/composer.json [hunk 5]
```diff
@@ -17,7 +17,7 @@
     ],
     "require": {
         "php": "^7.1.3",
-        "symfony/config": "^4.2|^5.0",
```
Comment (stof) [inline src/Symfony/Bundle/TwigBundle/composer.json]: this requirement should be kept, as we do depend on it (and we should not rely on the transitive dependency coming from elsewhere)

**Pull request merged**